### PR TITLE
Upgrade to Bleach 1.5, to limit link schemes to an allowlist #2860

### PR DESCRIPTION
**Pull request by SWAGATSWAROOP**
### Pull Request Checklist
Uncommented the line of allowed protocols from safe_markup function and Allowed protocols as support is added in bleach library.

* [x] Pull request is based on the develop branch
* [ ] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by SWAGATSWAROOP** [comment]
sign off: SWAGAT SWAROOP PARIDA swagatswroop@gmail.com

I am new to open source and this is my first contribution. If any mistake from my side forgive me and point out my mistake so that I can learn and grow with community.

**Comment by SWAGATSWAROOP** [comment]
Sorry Mistakenly Closed

**Comment by SWAGATSWAROOP** [comment]
sign off : SWAGAT SWAROOP PARIDA swagatswaroop@gmail.com

This is my first open source contribution. If there are mistakes point out It will help me learn and grow with community.

File: synapse/push/mailer.py [hunk 1]
```diff
@@ -890,7 +890,7 @@ def safe_markup(raw_html: str) -> Markup:
                 tags=ALLOWED_TAGS,
                 attributes=ALLOWED_ATTRS,
                 # bleach master has this, but it isn't released yet
-                # protocols=ALLOWED_SCHEMES,
+                protocols=ALLOWED_SCHEMES,
```
Comment (MadLittleMods) [inline synapse/push/mailer.py:893]: We still have a minimum version of [`bleach == 1.4.3`](https://github.com/matrix-org/synapse/blob/4ba528d9c31e1af75f3532ac2a52c1b12365f592/pyproject.toml#L193) specified which would also need to be updated. I don't know what the constraint is on why it's set so low at the moment. 

Relevant `bleach` 1.5 changelog that adds the `protocols`option : https://bleach.readthedocs.io/en/latest/changes.html#version-1-5-november-4th-2016

Comment (MadLittleMods) [inline synapse/push/mailer.py:893]: We're using [`bleach == 6.0.0` in `poetry.lock`](https://github.com/matrix-org/synapse/blob/4ba528d9c31e1af75f3532ac2a52c1b12365f592/poetry.lock#L203-L204) but it looks like there is a [breaking change where it needs to be a `Set` now](https://github.com/mozilla/bleach/blob/13d6c0c91d2b1bef52ad2b5a2812f76dda18298f/CHANGES#L9-L36). We can't really support both versions before and after as far as I can tell so would need to converge either way on our dependency versions

Comment (DMRobertson) [reply synapse/push/mailer.py:893]: No particular constraint AFAICS; see also https://github.com/matrix-org/synapse/pull/14906#discussion_r1086535472

Comment (DMRobertson) [reply synapse/push/mailer.py:893]: That's slightly alarming. I couldn't see a way in which this was obviously broken if you pass in a list:

```
$ python
Python 3.11.3 (main, May 24 2023, 00:00:00) [GCC 12.3.1 20230508 (Red Hat 12.3.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import bleach
>>> bleach.__version__
'6.0.0'
>>> bleach.clean()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: clean() missing 1 required positional argument: 'text'
>>> bleach.clean('aaaaa')
'aaaaa'
>>> bleach.clean('<naughty>')
'&lt;naughty&gt;'
>>> bleach.clean('<naughty><a>')
'&lt;naughty&gt;<a></a>'
>>> bleach.clean('<naughty><a href="foo">')
'&lt;naughty&gt;<a href="foo"></a>'
>>> bleach.clean('<naughty><a href="foo">', tags=['a'])
'&lt;naughty&gt;<a href="foo"></a>'
>>> bleach.clean('<naughty><a href="foo">', tags={'a'})
'&lt;naughty&gt;<a href="foo"></a>'
>>> bleach.clean('<naughty><a href="foo">', tags={'a', 'naughty'})
'<naughty><a href="foo"></a></naughty>'
>>> bleach.clean('<naughty><a href="foo">', tags={'naughty'})
'<naughty>&lt;a href="foo"&gt;</naughty>'
>>> bleach.clean('<naughty><a href="foo">', tags=['naughty'])
'<naughty>&lt;a href="foo"&gt;</naughty>'
```

**Pull request closed without merging**